Yoast/wordpress-seo
# Clean vendor_prefixed before running composer install

## Context


* Over the past few weeks, we ran into the situation multiple times that the `vendor_prefixed` folder in an artifact (=a zip) contained a `ruckusing` folder, while `ruckusing` was removed from the project weeks ago. The reason for this is that the `vendor_prefixed` folder is never cleaned. Therefore, we'd like to clean the `vendor_prefixed` folder every time `composer install` is run and `grunt artifact` is run.

## Summary


This PR can be summarized in the following changelog entry:

* Cleans the `vendor_prefixed` folder every time `composer install` and `grunt artifact` are run.

## Relevant technical choices:

* ~~Using the `pre-install-cmd` command event, [which fires](https://getcomposer.org/doc/articles/scripts.md) before the `install` command is executed with a lock file present.~~
* ~~Using the `rm -rf ./vendor_prefixed/*/` command to remove all the directories, but not the `.gitignore` file from `vendor_prefixed`.~~
* ~~The force flag (`-f`) prevents the script from crashing on a clean install where `vendor_prefixed` doesn't contain any directories.~~
* **Update:** the `pre-install-cmd` command broke the Travis build. It looked like a complicated timing issue that wasn't easy to solve. Because our main need is to have working zips, it was decided together with @herregroen to only focus on a fix for `grunt artifact`.
* I added a new `clean.js` command.
* No solution needed for the add-ons, they don't have `vendor_prefixed`.
* Unrelatedly to the problem this PR aims to solve, I removed the `"composer install"` alias from `shell.js`. It seemed to be an unused alternative to `"composer-install"` (with a dash). A [search](https://github.com/search?q=org%3AYoast+%27shell%3Acomposer+install%27&type=code) in the Yoast organization shows that `shell:composer install` isn't used anywhere.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**~~Test with `composer install`~~**
* ~~Add a directory to `vendor_prefixed`, for example an empty directory called `my-package`. Add a file to this directory.~~
* ~~Run `composer install`.~~
* ~~Inspect `vendor_prefixed`, it should contain its original packages but not `my-package`.~~
* ~~Remove all the directories from `vendor_prefixed`, but not `.gitignore`. This would be the situation you'd have if you had just cloned `wordpress-seo`.~~
* ~~Run `composer install` and see that `vendor_prefixed` gets filled with directories.~~

**Test with `grunt artifact`**
* Add a directory to `vendor_prefixed`, for example an empty directory called `my-package`. Add a file to this directory.
* Run `grunt artifact`. (You can stop the process when you're past the `composer install` step)
* Inspect `vendor_prefixed`, it should contain its original packages but not `my-package`.
* Remove all the directories from `vendor_prefixed`, but not `.gitignore`. This would be the situation you'd have if you had just cloned `wordpress-seo`.
* Run `grunt artifact`. (You can stop the process when you're past the `composer install` step)
* See that `vendor_prefixed` has been filled with directories.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* No need for QA to test this PR. No impact is expected, because the only thing we're doing in this PR is removing all the packages from `vendor_prefixed` before installing them again.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes P2-456
